provisioner/claim-reconciler/reconiler.go

```diff
@@ -81,33 +113,89 @@ func (r *objectBucketClaimReconciler) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, fmt.Errorf("%v", err)
 	}
 
-	// TODO wap in retry logic
-	if err = r.client.Create(context.TODO(), ob); err != nil {
-		return reconcile.Result{}, err
-	}
+	// TODO wrap in retry logic
+	err = wait.PollImmediate(r.retryInterval, r.retryTimeout, func() (done bool, err error) {
+		if err = r.client.Create(context.TODO(), ob); err != nil {
+			if apierrs.IsAlreadyExists(err) {
+				return true, fmt.Errorf("object bucket %s already exists: %v", ob.Name, err)
+			} else {
+				return true, fmt.Errorf("object bucket %s creation failed: %v", ob.Name, err)
+			}
+		}
+		return true, nil
+	})
+
+	credSecret := newCredentailsSecret(obc, s3keys)
+	err = wait.PollImmediate(r.retryInterval, r.retryTimeout, func() (done bool, err error) {
```

Comment:
probably want a more generic _create_this_resource func which handles retries, polling, etc for Secrets, CMs, and OBs.